Chat

d-u-d-e opened the pull request:
Aggiunto le liste per vedere i match creati e a cui si è iscritti. La chat dovrebbe essere funzionante. Non sono sicuro ma potrebbero capitare eccezioni di index out of bound. Ho testato più volte e non sembrano ripetersi più. Probabilmente erano dovuti a dati non coerenti nel database.

d-u-d-e committed:
reading messages related to a match should now work

d-u-d-e committed:
fixed layout for messages

d-u-d-e committed:
fixed names and added method to Message class

Commit:
new file:   app/src/main/java/com/es/findsoccerplayers/ChatActivity.java
	new file:   app/src/main/res/layout/activity_chat.xml

Commit:
new file:   app/src/main/java/com/es/findsoccerplayers/User.java

d-u-d-e committed:
fixed unresolved conflicts

d-u-d-e committed:
completed rebase with master

d-u-d-e committed:
refactoring

d-u-d-e committed:
setting up fragments

d-u-d-e committed:
ultimating changes

d-u-d-e committed:
fixes

d-u-d-e committed:
more fixes and almost ultimated everything

kzmdti committed:
bug fixed

d-u-d-e committed:
tried to solve issue with message username not displaying correctly

d-u-d-e committed:
set item animator to null for recycler view

this line is mandatory, see a known android bug at: https://stackoverflow.com/questions/35653439/recycler-view-inconsistency-detected-invalid-view-holder-adapter-positionviewh

lucagiordano-visia committed:
Modifica stile messaggio chat + creazione activity infoBookedMatch

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/Utils.java:
```diff
@@ -82,6 +82,7 @@ public void onSuccess(Void aVoid) {
                 Log.w(tag, "Database successfully updated with new general match info");
             }
         });
+        return true;
```
questo non serve molto, tipo ritorna sempre true.

lucagiordano-visia replied:
Mi serve perchè su create match se la funzione torna true mostra un toast di successo. E' da implementare la casistica in cui fallisce

d-u-d-e replied:
eh si, vabbeh lasciamo cosi per ora, metti un TODO

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/ActivityCreateMatch.java:
```diff
@@ -135,7 +135,11 @@ public void onClick(View v) {
+                        Toast.makeText(ActivityCreateMatch.this, "Match successfully created", Toast.LENGTH_SHORT).show();
+                        startActivity(new Intent(ActivityCreateMatch.this, ActivityMain.class));
+                        finish();
+                    }
```
usa il flag reorder to front. Tanto esiste già la main activity.

d-u-d-e commented on app/src/main/res/layout/frag_chat_message_left.xml:
```diff
@@ -20,6 +20,7 @@
             android:layout_alignParentTop="true"
             android:layout_marginStart="5dp"
             android:text="username"
+            android:textStyle="italic"
```
bold?

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/fragments/FragmentInfoBookedMatch.java:
```diff
@@ -4,16 +4,78 @@
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.frag_info_booked_match, container, false);
+
+        field = view.findViewById(R.id.frag_info_booked_match_campoTV);
```
nomi sempre in inglese nel layout, meglio farlo subito che se no ci tocca riguardare tutto da 0 alla fine

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/fragments/FragmentInfoBookedMatch.java:
```diff
@@ -4,16 +4,78 @@
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                m = new Match();
+                m = dataSnapshot.getValue(Match.class);
```
non serve che crei un'istanza di Match(), te la ritorna getValue

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/fragments/FragmentInfoBookedMatch.java:
```diff
@@ -4,16 +4,78 @@
+                field.setText(m.getPlaceName().toString());
+                day.setText(m.getMatchData().toString());
+                time.setText(m.getMatchHour().toString());
+                quota.setText("----");
```
quota in inglese

lucagiordano-visia committed:
Modifiche review

kzmdti commented on app/src/main/java/com/es/findsoccerplayers/fragments/FragmentInfoBookedMatch.java:
```diff
@@ -4,18 +4,76 @@
+                time.setText(m.getMatchHour().toString());
+                money.setText("----");                                              //TODO
+                missingPlayers.setText(Integer.toString(m.getPlayersNumber()));
+                description.setText(m.getDescription().toString());
```
il toString() è ridondante. Gia getPlaceName(), getMatchData etc ritornano una Stringa

d-u-d-e replied:
vero

kzmdti commented on app/src/main/java/com/es/findsoccerplayers/adapter/MessageAdapter.java:
```diff
@@ -0,0 +1,80 @@
+        holder.message.setText(m.getText());
+
+        String dateStr = DateUtils.formatDateTime(context, m.getTimestamp(), DateUtils.FORMAT_ABBREV_MONTH |
+                DateUtils.FORMAT_SHOW_DATE | DateUtils.FORMAT_SHOW_TIME);
```
l'ora sarebbe meglio se mostra in formato 24H

lucagiordano-visia replied:
E' già nel formato 24h

kzmdti commented on app/src/main/java/com/es/findsoccerplayers/ActivityInfoBookedMatch.java:
```diff
@@ -16,16 +17,20 @@
-        adapter.addFragment(new FragmentInfoBookedMatch(), "INFO");
-        adapter.addFragment(new FragmentChat(), "CHAT");
+        adapter.addFragment(new FragmentInfoBookedMatch(matchID), "INFO");
+        adapter.addFragment(new FragmentChat(matchID), "CHAT");
```
CHAT e INFO vanno come resource R.string....

lucagiordano-visia replied:
se lo faccio la risorsa gli viene passata con un int, e non stampa la stringa, ma il codice associata alla stringa

kzmdti replied:
usa getString(R.string....) e ti passa la stringa

kzmdti commented on app/src/main/java/com/es/findsoccerplayers/ActivityInfoBookedMatch.java:
```diff
@@ -16,16 +17,20 @@
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
 
+        Intent i = getIntent();
+        String matchID = i.getStringExtra("match");
```
match mettiamolo come 
private static final String MATCH = "match";
Questo anche in tutte le altre activity e Fragment

lucagiordano-visia replied:
perchè?

kzmdti replied:
perche il prof ci tiene. L'ha detto anche a lezione che le chiavi siano definite come static final. Buona programazzione. So che è una rottura di palle, ma ci tiene

lucagiordano-visia committed:
modifiche suggerite

lucagiordano-visia committed:
Modifica elenco match

- Creazione elemento match per la lista dei match
- Modifica classi di MatchAdapter e FragmentBookedMatches per risolvere bug e implementare l'elemento match della lista

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/Utils.java:
```diff
@@ -88,4 +89,110 @@ public void onSuccess(Void aVoid) {
     }
+
+    public static String getPreviewDescription(String description){
+        if(description.length()<20)
```
ci va <= perché sotto prendi 21 caratteri

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/Utils.java:
```diff
@@ -88,4 +89,110 @@ public void onSuccess(Void aVoid) {
+            return (description.substring(0,21)+"...");
+        }
+    }
+
```
perché hai scritto questi due metodi? Usa `Calendar.getDisplayName`

lucagiordano-visia replied:
Quel metodo non mi pare faccia quello che interessa a me

kzmdti replied:
Calendar non va bene. Ma potresti usare SimpleDateFormat di Java che ti da anche i nomi nella lingua desiderata
https://www.mrwebmaster.it/java/simpledateformat-formattare-data-java_11616.html

d-u-d-e replied:
si fa esattamente questo, ti estrae i field: https://docs.oracle.com/javase/7/docs/api/java/util/Calendar.html#getDisplayName(int,%20int,%20java.util.Locale)

d-u-d-e replied:
puoi anche scegliere se visualizzare in formato lungo o corto

lucagiordano-visia replied:
sinceramente non ho ben capito come implementarlo, magari potete farlo voi se avete già un'idea di come fare

d-u-d-e commented on app/src/main/java/com/es/findsoccerplayers/fragments/FragmentBookedMatches.java:
```diff
@@ -27,27 +30,40 @@
+                matches.get(position);
+                Intent i = new Intent(getContext(), ActivityInfoBookedMatch.class);
+                i.putExtra("match", matches.get(position).getMatchID());
+                i.setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
```
perché il flag?

lucagiordano-visia replied:
Non è il flag per tenere in memoria l'activity senza che venga ricreata?

d-u-d-e replied:
no quel flag serve a startare l'activity già presente nello stack

lucagiordano-visia committed:
modifiche
